tests/docs/smoke-all/crossrefs/theorem/theorem-1.qmd

```diff
@@ -0,0 +1,44 @@
+---
+keep-typ: true
```

Comment:
Is `keep-typ` necessary here (and in the other file as well)?

I seem to remember that we changed our tests to do that automatically when the test spec includes a `typst` entry (similarly for pdf). Another way that we achieve this is by changing `output-ext` in format. This is a bit of magic right now, but

```
format:
  typst:
    output-ext: typ
```

should also make it so that Quarto stops at `typ` rather than producing a PDF file.

Reply:
I didn't know about `output-ext`, changing the two tests to use it.

The tests fail with `No such file or directory` otherwise...

Reply:
> I didn't know about `output-ext`, changing the two tests to use it.

Originally it's there to allow `.html` -> `.htm` hacks and so on, but we've now added puns for latex and typst where if you use `tex` or `typ` respectively, we don't actually call the final output recipe.

> The tests fail with No such file or directory otherwise...

That makes sense.

Reply:
I think you remember that because we had the discussion on slack when you added `ensureTypstFileRegexMatches`

Using `ensureTypstFileRegexMatches` requires `keep-typ: true` . You added a comment to remember also

https://github.com/quarto-dev/quarto-cli/blob/3dcd0d11842d94686a3607c95dbbe3a842d3e83c/tests/verify.ts#L238-L250

It will match `.pdf` file, and look for the `.typ` file with same name. 

Before this addition, when I did the first typst test I just used  `output-ext: typ` and `ensureFileRegexMatches()` 

`ensureTypstFileRegexMatches` will run a PDF conversion so it is better to test that the .typ produce render to `.pdf`, but when this is not needed the other way can be used. Though running Typst is quite fast. 

Hope it helps clarify the situation (which is not the best but works ok)